mathjax3-ts/output/chtml/Wrappers/mn.ts

```diff
@@ -0,0 +1,61 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2018 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements the CHTMLmn wrapper for the MmlMn object
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {CHTMLWrapper} from '../Wrapper.js';
+import {CHTMLWrapperFactory} from '../WrapperFactory.js';
+import {MmlMn} from '../../../core/MmlTree/MmlNodes/mn.js';
+import {MmlNode, AbstractMmlNode, TextNode} from '../../../core/MmlTree/MmlNode.js';
+
+/*****************************************************************/
+/*
+ * The CHTMLmn wrapper for the MmlMn object
+ *
+ * @template N  The HTMLElement node class
+ * @template T  The Text node class
+ * @template D  The Document class
+ */
+export class CHTMLmn<N, T, D> extends CHTMLWrapper<N, T, D> {
+    public static kind = MmlMn.prototype.kind;
+
+    /*
+     * @override
+     */
+    public remapChars(chars: number[]) {
+        //
+        //  Convert a leading hyphen to a minus
+        //
+        if (chars.length) {
+            const string = this.font.getRemappedChar('mn', chars[0]);
+            if (string) {
+                const c = this.unicodeChars(string);
+                if (c.length === 1) {
```

Comment:
Use  x? y : z ;

Reply:
I don't quite see what you have in mind, here.  Do you want this?

              c.length === 1 ? (chars[0] = c[0]) : (chars = c.concat(chars.slice(1)));

In the (most common) case that the `string` is a single character, it is more efficient to simply replace the initial character rather than build a new array, as in the `else` clause.  Otherwise, one could simply always do the `chars.concat(chars.slice(1))`.  I just didn't want to have to construct two additional arrays just to replace the first character.

If this isn't what you had in mind, can you be more explicit?

Reply:
The code snippet was what I had in mind. Sorry, I did not express it better.
I agree on the efficiency of replacing. And having another look, I see that my suggestion is confusing, as the return type of the conditional would be ambiguous: `number` vs `Array`.
So please leave unchanged.

Reply:
OK, sounds good.